src/test/java/org/springframework/samples/petclinic/cache/GenericEntityCacheIntegrationTest.java

```diff
@@ -0,0 +1,662 @@
+                    public void mockFindAll(Object repository, Collection<Object> entities) {
+                        when(((VetRepository) repository).findAll()).thenReturn((Collection) entities);
+                    }
+                    @Override
+                    public void mockFindById(Object repository, Integer id, Object entity) {
+                        when(((VetRepository) repository).findById(id)).thenReturn((Vet) entity);
+                    }
+                    @Override
+                    public void mockSave(Object repository, Object entity) {
+                        doNothing().when((VetRepository) repository).save((Vet) entity);
+                    }
+                    @Override
+                    public void mockDelete(Object repository, Object entity) {
+                        doNothing().when((VetRepository) repository).delete((Vet) entity);
+                    }
+                })
+                .build();
+        } else if ("PetType".equals(entityName)) {
+            return EntityCacheTestConfiguration.<Object, Object>builder()
+                .entityName("PetType")
+                .apiEndpoint("/pettypes")
+                .entityFactory(builder -> createPetType(builder.getId(), builder.getName()))
+                .dtoFactory(builder -> createPetTypeDto(builder.getId(), builder.getName()))
+                .repository(petTypeRepository)
+                .expectedCreateStatus(HttpStatus.CREATED)
+                .expectedUpdateStatus(HttpStatus.NO_CONTENT)
+                .expectedDeleteStatus(HttpStatus.NO_CONTENT)
+                .repositoryOperations(new EntityCacheTestConfiguration.RepositoryOperations<Object>() {
+                    @Override
+                    public void mockFindAll(Object repository, Collection<Object> entities) {
+                        when(((PetTypeRepository) repository).findAll()).thenReturn((Collection) entities);
+                    }
+                    @Override
+                    public void mockFindById(Object repository, Integer id, Object entity) {
+                        when(((PetTypeRepository) repository).findById(id)).thenReturn((PetType) entity);
+                    }
+                    @Override
+                    public void mockSave(Object repository, Object entity) {
+                        doNothing().when((PetTypeRepository) repository).save((PetType) entity);
+                    }
+                    @Override
+                    public void mockDelete(Object repository, Object entity) {
+                        doNothing().when((PetTypeRepository) repository).delete((PetType) entity);
+                    }
+                })
+                .build();
+        }
+        throw new IllegalArgumentException("Unknown entity: " + entityName);
+    }
+
+    /**
+     * Individual test demonstrating Vet cache testing with the framework
+     */
+    @Test
+    void testVetCacheWithGenericFramework() {
```

Comment:
Sometimes test cases are parametrized, sometimes not, probably it's better to follow same style.